### PR TITLE
Add a more complete Teleport Cloud introduction

### LINKED ISSUE #11226: Expand Teleport Cloud intro
## Details
Provide a expanded Teleport Cloud intro that provides the similar level of content as other sections.   

### Category
- Improve Existing

### PR DESCRIPTION
**Pull request by ptgott**
Fixes #11226

- Include more detail about what Teleport Cloud encompasses
- Explain the benefits of Teleport Cloud
- Add Tiles to link to all pages in the /docs/cloud section

### CODE REVIEW

File: docs/pages/cloud/introduction.mdx [hunk 1]
```diff
@@ -1,16 +1,60 @@
...
-- Get started with cloud [in 5 minutes](./getting-started.mdx).
-- Join the [Teleport Discussons](https://github.com/gravitational/teleport/discussions) and ask a question.
-- Join the [Slack channel](https://goteleport.com/slack).
+The following management tasks are handled for you:
+
+- Auth and Proxy Service upgrades, security patches, scaling, monitoring,
+  maintenance, and network configuration
+- Auth Service backend management
+- DNS record management
+- Certificate authority rotation
```
Comment (zmb3) [inline docs/pages/cloud/introduction.mdx:20]: Is this true?

**Review by zmb3 — APPROVED** [review]

File: docs/pages/cloud/introduction.mdx (hunk 1 above)
Comment (ptgott) [reply docs/pages/cloud/introduction.mdx]: I should have checked this more carefully—looks like `tctl auth rotate` is in the hands of the user. In this case, I was thinking of the TLS credentials that the Proxy Service uses to communicate with clients, though I'm not sure that's worth mentioning here.

File: docs/pages/cloud/introduction.mdx [hunk 2]
```diff
@@ -1,16 +1,59 @@
 ---
-title: Cloud
-description: Teleport as a service
-h1: Teleport cloud
+title: Teleport Cloud
+description: Teleport Cloud is a fully managed deployment of Teleport.
```
Comment (klizhentas) [inline docs/pages/cloud/introduction.mdx:3]: To be more precise: is a deployment of Teleport control plane managed by Teleport Cloud team

File: docs/pages/cloud/introduction.mdx [hunk 3]
```diff
@@ -1,16 +1,59 @@
...
 
-- Get started with cloud [in 5 minutes](./getting-started.mdx).
-- Join the [Teleport Discussons](https://github.com/gravitational/teleport/discussions) and ask a question.
-- Join the [Slack channel](https://goteleport.com/slack).
+The following management tasks are handled for you:
+
+- Auth and Proxy Service upgrades, security patches, scaling, monitoring,
+  maintenance, and network configuration
+- Auth Service backend management
+- DNS record management
```
Comment (klizhentas) [inline docs/pages/cloud/introduction.mdx:19]: Also backup and restore, audit storage

**Review by klizhentas — APPROVED** [review]
Looks good with a couple of nits

**Pull request merged**